File: 2021/day-09/main_test.py
import unittest

import main


class TestCase(unittest.TestCase):
    def setUp(self):
        self.input = main.read_input('input_test.txt')

    def test_part_one(self):
        self.assertEqual(main.part_one(self.input), 15)

    def test_part_two(self):
        self.assertEqual(main.part_two(self.input), 1134)

File: 2022/day-02/main_test.go
package main

import "testing"

func TestPartOne(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partOne(input)
	want := 15
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}

func TestPartTwo(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partTwo(input)
	want := 12
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}
